### ISSUE TITLE
Division by zero when trying to generate a transform with only quality specified

### ISSUE THREAD

**Issue by OscarBarrett** [craftcms/cms#5590]
### Description
According to the [Image Transform docs](https://docs.craftcms.com/v3/image-transforms.html#possible-values) `width`/`height` may be omitted when generating transforms. However when attempting to generate a transform via a GraphQL query with only quality specified, it throws a division by zero error.

In addition, if `immediately` is set to true then the returned value for the url is `null` whereas the return type of `AssetInterface.url` is specified as `String`.

I haven't tested if this also affects transforms from twig.

### Steps to reproduce

1. Set up an Asset field, add it to the field layout of something and upload an image to the field.
2. Make a GraphQL request for the asset with only `quality` set
```
{
  globalSets(handle: "general") {
    ...on general_GlobalSet {
      image {
        url(quality: 30)
      }
    }
  }
}
```
3. Follow the transform link - an exception is thrown.

```
yii\base\ErrorException: Division by zero in /app/vendor/craftcms/cms/src/image/Raster.php:245
Stack trace:
#0 /app/vendor/craftcms/cms/src/web/ErrorHandler.php(80): yii\base\ErrorHandler->handleError(2, 'Division by zer...', '/app/vendor/cra...', 245)
#1 /app/vendor/craftcms/cms/src/image/Raster.php(245): craft\web\ErrorHandler->handleError(2, 'Division by zer...', '/app/vendor/cra...', 245, Array)
#2 /app/vendor/craftcms/cms/src/services/AssetTransforms.php(1442): craft\image\Raster->scaleToFit(0, 0)
#3 /app/vendor/craftcms/cms/src/services/AssetTransforms.php(710): craft\services\AssetTransforms->_createTransformForAsset(Object(craft\elements\Asset), Object(craft\models\AssetTransformIndex))
#4 /app/vendor/craftcms/cms/src/services/AssetTransforms.php(613): craft\services\AssetTransforms->_generateTransform(Object(craft\models\AssetTransformIndex))
#5 /app/vendor/craftcms/cms/src/controllers/AssetsController.php(885): craft\services\AssetTransforms->ensureTransformUrlByIndexModel(Object(craft\models\AssetTransformIndex))
#6 [internal function]: craft\controllers\AssetsController->actionGenerateTransform(34)
#7 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#8 /app/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#9 /app/vendor/craftcms/cms/src/web/Controller.php(187): yii\base\Controller->runAction('generate-transf...', Array)
#10 /app/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('generate-transf...', Array)
#11 /app/vendor/craftcms/cms/src/web/Application.php(299): yii\base\Module->runAction('assets/generate...', Array)
#12 /app/vendor/craftcms/cms/src/web/Application.php(565): craft\web\Application->runAction('assets/generate...', Array)
#13 /app/vendor/craftcms/cms/src/web/Application.php(278): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#14 /app/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#15 /app/web/index.php(21): yii\base\Application->run()
#16 {main}
```

### Additional info

- Craft version: 3.3.20.1
- PHP version: 7.3.11
- Database driver & version: MySQL 5.7
- Plugins & versions: N/A

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just fixed for the next 3.4 release.

> In addition, if `immediately` is set to true then the returned value for the url is `null` whereas the return type of `AssetInterface.url` is specified as `String`.

That `immediately` param isn’t really needed anymore in 3.4 since GraphQL always generates transforms immediately now. If you’re still having a similar issue after updating to 3.4, please post as a new issue.

**Comment by brandonkelly** [comment]
Craft 3.4.4 is out now with that fix.
